File: play/template.rs
/* 
 * Author: Canis Lupus Loner
 * Compile and run with cargo or rustc
 */

// #![warn(unused_macros)]
// #![warn(unused_imports)]
// #![warn(unused_must_use)]
// #![allow(unused_variables)]

// Standard Library //
use std::io;
use std::fs;

use std::io::Write;

use std::cmp::{min, max, Ordering};

// use std::collections::{BinaryHeap, VecDeque};
// use std::collections::{BTreeMap, BTreeSet];
// use std::collections::hash_map::HashMap;
// use std::collections::hash_set::HashSet;

// Macro //
macro_rules! input {
    ($s: expr, $t: ty) => {
        $s.next().unwrap().parse::<$t>().unwrap()
    };

    ($s: expr, String) => {
        $s.next().unwrap().to_string()
    };

    ($s: expr, $t: ty, $n : expr) => {
        (0..$n)
            .map(|_| $s.next().unwrap().parse::<$t>().unwrap())
            .collect::<Vec<$t>>()
    };
}

// External //


// Global //


// Functions //


fn main() -> io::Result<()> {
    #[cfg(not(LOCAL))]
    let stdin: String = io::read_to_string(io::stdin().lock()).unwrap();
    #[cfg(any(LOCAL, feature = "LOCAL"))]
    let stdin: String = fs::read_to_string("input.txt").unwrap();

    #[cfg(not(LOCAL))]
    let stdout  = io::stdout().lock();
    #[cfg(any(LOCAL, feature = "LOCAL"))]
    let stdout = fs::File::create("output.txt").unwrap();
    
    let mut cin = stdin.split_whitespace();
    let mut cout = stdout;
    
    // Solution //
    let t: usize = 1; // for single test
    // let t: usize = input!(cin, usize); // for multiple tests
    for _ in 0..t {
        // Test Case //
    }

    dbg!("Everyone is awesome!");
    return Ok(());
}

File: play/template.cpp
/* 
 * Author: Canis Lupus Loner
 */


 #pragma GCC optimize("O3", "fast-math", "unroll-loops")

/*
 * Compile and Run
 * g++ -std=c++2a -DLOCAL -Wall -Wextra -pedantic 
 * -g -Wconversion -Wshadow <source> -I . -o <binary>
 * ./<binary>
 * gdb <binary> // on failure.
 */

// All STL
#include <bits/stdc++.h>

// Individual STL Libs
// I/O Libs
#include <cstdio> // printf and scanf
#include <fstream> // files nad buffers
#include <iomanip> // setprecision
#include <iostream> // standard input output, <istream> + <ostream>
// #include <print> // formatting options
#include <sstream>
// Math Libs
#include <bit> // popcount, rotate left/right, bit floor/ceil, etc.
#include <cmath> // abs, trig, log, etc.
#include <complex> // complex numbers
// #include <linalg> // linear algebra
// #include <numbers> // e, pi, phi, etc
#include <numeric> // iota, accumulate, gcd, lcm, midpoint, inner product, etc.
#include <random> // generate random numbers from distributions
// Algo Libs
#include <algorithm> // fill, rotate, sort, swap, search, heap, permutation, etc.
// Limits
// #include <climits> // INT_MIN/MAX, etc.
#include <limits> // numeric limits and functions
// Utility
#include <utility> // pair, tuple, compare, etc.
// #include <tuple>
#include <functional>
#include <any> // like python variable
#include <bitset> // fixed length array of bits
// #include <cstdlib> // abs, rand, sort, search for C
// Containers
#include <array> // like fixed sized arrays in C but with member functions
#include <vector>
#include <queue>
#include <deque> // // double ended queue
#include <list> // doubly linked list
#include <stack>
#include <set> // includes multiset
#include <unordered_set> // hashing and chaining
#include <map> // rb-tree
#include <unordered_map> // hashing and chaining
// String Libs
#include <string>
// #include <cstring> // for C stype operations: strcpy, strcat
// Other Libs
#include <iterator>
#include <regex>

using namespace std;

typedef long double ld;
typedef long long int ll;

// Macro //
#define X first
#define Y second
#define pll pair<ll, ll>
#define pii pair<int, int>
#define all(x) (x).begin(), (x).end()
#define debug(x) cerr << x << endl

// External //

/*
 * Citation
 */

// // // // //

// Global //


// Functions //


// Main //
int main(int argc, char const *argv[]) {
    assert(sizeof(*argv) == sizeof(void*) * argc);

    // ios_base::sync_with_stdio(false);
    // cin.tie(NULL);

    // cout << fixed << setprecision(16);

    #ifndef ONLINE_JUDGE
    // freopen("input.txt", "r", stdin);
    // freopen("output.txt", "w", stdout);
    #else
    // online submission
    #endif

    #ifdef LOCAL
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
    #else
    // online submission
    #endif
    
    /* code */
    cerr << "On a starry moonlit night,\n"
    << "A lonely wolf begins to howl.\n"
    << "Wondering if his solutions are right,\n"
    << "'Do submit; maybe it's not all foul.'" << endl;
    
    return 0;
}


// Definitions //
